lnurl/core.py

```diff
@@ -23,82 +23,83 @@ def encode(url: str) -> Lnurl:
+async def get(url: str, *, response_class: Optional[Any] = None) -> LnurlResponseModel:
+    async with httpx.AsyncClient() as client:
+        try:
+            res = await client.get(url)
+            res.raise_for_status()
+        except Exception as e:
+            raise LnurlResponseException(str(e))
 
-    if response_class:
-        assert issubclass(response_class, LnurlResponseModel), "Use a valid `LnurlResponseModel` subclass."
-        return response_class(**req.json())
+        if response_class:
+            assert issubclass(response_class, LnurlResponseModel), "Use a valid `LnurlResponseModel` subclass."
+            return response_class(**res.json())
 
-    return LnurlResponse.from_dict(req.json())
+        return LnurlResponse.from_dict(res.json())
 
 
-def handle(
+async def handle(
     bech32_lnurl: str,
     response_class: Optional[LnurlResponseModel] = None,
-    verify: Union[str, bool] = True,
 ) -> LnurlResponseModel:
     try:
         if "@" in bech32_lnurl:
             lnaddress = LnAddress(bech32_lnurl)
-            return get(lnaddress.url, response_class=response_class, verify=verify)
+            return await get(lnaddress.url, response_class=response_class)
         lnurl = Lnurl(bech32_lnurl)
     except (ValidationError, ValueError):
         raise InvalidLnurl
 
     if lnurl.is_login:
         return LnurlAuthResponse(callback=lnurl.url, k1=lnurl.url.query_params["k1"])
 
-    return get(lnurl.url, response_class=response_class, verify=verify)
+    return await get(lnurl.url, response_class=response_class)
 
 
-def execute(bech32_or_address: str, value: str) -> LnurlResponseModel:
+async def execute(bech32_or_address: str, value: str) -> LnurlResponseModel:
     try:
         res = handle(bech32_or_address)
     except Exception as exc:
         raise LnurlResponseException(str(exc))
 
     if isinstance(res, LnurlPayResponse) and res.tag == "payRequest":
-        return execute_pay_request(res, value)
+        return await execute_pay_request(res, value)
     elif isinstance(res, LnurlAuthResponse) and res.tag == "login":
-        return execute_login(res, value)
+        return await execute_login(res, value)
 
     raise LnurlResponseException(f"{res.tag} not implemented")  # type: ignore
 
 
-def execute_pay_request(res: LnurlPayResponse, msat: str) -> LnurlResponseModel:
-    print(res)
+async def execute_pay_request(res: LnurlPayResponse, msat: str) -> LnurlResponseModel:
     if res.min_sendable > MilliSatoshi(msat) > res.max_sendable:
         raise LnurlResponseException(f"Amount {msat} not in range {res.min_sendable} - {res.max_sendable}")
     try:
-        req = requests.get(
-            res.callback,
-            params={
-                "amount": msat,
-            },
-        )
-        req.raise_for_status()
-        return LnurlResponse.from_dict(req.json())
+        async with httpx.AsyncClient() as client:
+            res2 = await client.get(
```

Comment:
nitpick: do not use `res2`. Maybe `lnurl_res`& `res`, or `res` & `r`

Reply:
its the second response though :) `res` is the first response `LnurlPayResponse`